test/e2e/e2e_agent_test.go

```diff
@@ -382,7 +383,11 @@ func (s *plannerService) GetSource(id uuid.UUID) (*api.Source, error) {
 
 	res, err := s.c.GetSourceWithResponse(ctx, id)
 	if err != nil || res.HTTPResponse.StatusCode != 200 {
-		return nil, fmt.Errorf("Error listing sources")
+		code := -1
+		if res != nil {
```

Comment:
I think it can't be nil, because in upper level if we already access the `res` var